src/main/java/com/github/manolo8/darkbot/gui/drawables/DynamicEntitiesDrawer.java

```diff
@@ -189,11 +214,20 @@ private void drawHeroTarget(MapGraphics mg) {
         }
     }
 
-    private void drawEntity(MapGraphics mg, Entity entity, double size, boolean fill) {
+    private boolean isMemberVisible(GroupMember member) {
+        for (Player player : players) {
```

Comment:
I don't particularily like the o(n^2) complexity of this (for each group member you're iterating all the visible players). Consider instead creating a set with the ids of all group members, and when rendering players call remove on the set. After that when rendering the extra group members you just gotta check that they're in the set to know they are not-visible.


Roughly:
```
Set<Long> nonVisibleMembers = group.members().map(toIds).collect(toSet());
for (Player player : players) {
  // render the player
  nonVisibleMembers.remove(player.id());
}

for (GroupMember member : group.members()) {
  if (!nonVisibleMembers.contains(member)) continue;
  // render ghostly-figure of the member
}
```